### ISSUE TITLE
Ignore variable declaration if it's the zero value

### ISSUE THREAD

**Issue by natefinch** [sanposhiho/wastedassign#39]
At work, we are working on updating all our linting, and one of the problems we have hit with wastedassign is that code like this gets flagged:

```go
isFoo := false
// ...

isFoo, err := bar.Baz()
```
 Now, I know that you should probably write `var isFoo bool` ... but it's not technically incorrect, and almost assuredly does not indicate an error, just someone who is used to other languages. It would be great if there were a way to avoid flagging that kind of a line so that we can use it for the other cases when it is much more likely to indicate a bug.

**Comment by ericleb010** [comment]
I agree, and the error returned makes it a bit confusing to understand what the issue is exactly. Especially if someone isn't familiar with relying on a primitive's zero value.

**Comment by egonelbre** [comment]
I hit this as well with:
```
buildType := ""
```

**Comment by drshriveer** [comment]
This really would be great to fix. I personally use `name := defaultValue` over `var name type` because it makes it exceptionally clear what the start value is. If nothing else, I'd love a configuration flag for this.

**Comment by jadegopher** [comment]
Agreed with @drshriveer.
Declaration `err := error(nil)` is similar to `var err error` and I personally prefer first option but linter tells that is wasted assign.

**Comment by nirs** [comment]
I think this is a bug in the linter since both forms should be compiled to the same thing:

```
var a bool
b := false
```

If they no not compile to the same thing, this is a bug in the Go compiler.

Reporting this as an issue is harmful instead of helping. This just increase the chance that people will disable this linter and missing good reports that should be fixed.
